src/LowercaseSentenceTokenizer.java

```diff
@@ -29,8 +30,46 @@ public class LowercaseSentenceTokenizer implements Tokenizer {
    * @return a list of tokens, where each token is a word or a period
    */
   public List<String> tokenize(Scanner scanner) {
-    // TODO: Implement this function to convert the scanner's input to a list of words and periods
-    return null;
+    // DONE: Implement this function to convert the scanner's input to a list of words and periods
+    List <String> tokens = new ArrayList<>(); // creating an Arraylist
+
+    // reading the input until there's no text left to read from the Scanner
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine(); // saving the read input line in the variable
+      String words[] = line.split(" "); // adding spaces
```

Comment:
This works well! However, `Scanner` by default splits on whitespace. Can you think of how you could use `scanner.next` instead of `scanner.nextLine` to simplify your code?

Reply:
Thank you for the feedback! Instead of reading the whole line, I can read the words by simply using `scanner.next` . 

Possible Solution: 
`while (scanner.hasNext()) {
    String word = scanner.next(); 
}`